src/query/execute_schema.c

```diff
@@ -9031,9 +9052,17 @@ do_recreate_renamed_class_indexes (const PARSER_CONTEXT * parser, const char *co
   /* add indexes */
   for (saved = index_save_info; saved != NULL; saved = saved->next)
     {
+      SM_INDEX_STATUS old_status = SM_NORMAL_INDEX;
+
+      error = sm_get_constr_index_status (classmop, saved->name, &old_status);
```

Comment:
I don't think this works because indexes are dropped and only `SM_NORMAL_INDEX` will be returned.
You may have to consider to extend `SM_CONSTRAINT_INFO` or have a new struct to save the existing constraints.